### PR TITLE
improve(build): update pkg version and sign mac executables

### PR DESCRIPTION
**Pull request by edvald**
This is a precursor to updating to Node.js 14 and adding proper support for ARM64 builds.

No rush to merge this ahead of tomorrow's release btw @thsig.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -775,6 +775,35 @@ jobs:
           command: |
             sudo -E ./bin/garden plugins local-kubernetes cleanup-cluster-registry --root core/test/data/test-projects/container --env cluster-docker
 
+  test-macos:
+    macos:
+      xcode: "12.5.1"
+    steps:
+      - checkout
+      - *attach-workspace
+      # - run:
```
Comment (thsig) [inline .circleci/config.yml:784]: Remove comment?

File: cli/bin/garden [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env node --max-old-space-size=4096
```
Comment (thsig) [inline cli/bin/garden:1]: Is this a deliberate reversion of the changes introduced in https://github.com/garden-io/garden/pull/2812/files?

Comment (edvald) [reply cli/bin/garden:1]: It's not a reversion, it moves the option flag to `bin/garden` instead. This didn't work for some reason when calling that (which btw does not affect the dist build).

File: .circleci/config.yml (hunk 1 above)
Comment (edvald) [reply .circleci/config.yml:784]: Doesn't really matter, figure we might need this later.

**Review by thsig — APPROVED** [review]

**Pull request merged**